artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -1674,12 +1674,8 @@ public ServerSession createSession(final String name,
                                       final boolean autoCreateQueues,
                                       final OperationContext context,
                                       final Map<SimpleString, RoutingType> prefixes,
-                                      final String securityDomain) throws Exception {
-      String validatedUser = "";
-
-      if (securityStore != null) {
-         validatedUser = securityStore.authenticate(username, password, connection, securityDomain);
-      }
+                                      final String securityDomain,
+                                      String validatedUser) throws Exception {
```

Comment:
cool anticipating the authentication, if securityStore != null and validatedUser == null we could call `securityStore.authenticate` or thwroing a security exception

Reply:
I think that should go via server.validateUser, it seems securityStore == null only happens on shutdown. But best to utilise the same method as other protocol heads.